### ISSUE TITLE
Incorrect `reportAny` when type hinting `json.loads()` return value

### ISSUE THREAD

**Issue by Sevarin1** [DetachHead/basedpyright#1470]
### Description

In short, basedpyright claims the return value of `json.loads(valid_json_as_string)` is `Any`, when it is in fact a `dict`.
pyright (and also mypy) correctly accepts the detailed `dict` type hint and reports no issues.

basedpyright playground [link](https://basedpyright.com/?typeCheckingMode=all&code=JYWwDg9gTgLgBAKwM4QHYChMBMCGMcBccSMUcAvHAEQ1wDe6cTVARgDYQDGA1lgKYA3YJz5IqRANqMmMhjPnNUOEH3HUA9PwHqAHgEYqAGmkLmAMxIBPMKqJU%2BOmABYjJ01STAAXrep6AzAB0AKwA4q6mMlQgEACuqDCQwAlqAHJofG5MAL7Gpln0BVFKKmpUmoK6AEwRkdQWMNa%2BOhZ5dR7evlQAnP56IeFtkdFxCUkpROmomabZJgC66HM0VJhgOFBIfFhEWMIwEiRQhnBswCQSe5wHRydHcAA%2BcFN8828UiCiogRw4WEgACiQ5Fw%2BAAlEA) 

pyright playground [link](https://pyright-play.net/?code=JYWwDg9gTgLgBAKwM4QHYChMBMCGMcBccSMUcAvHAEQ1wDe6cTVARgDYQDGA1lgKYA3YJz5IqRANqMmMhjPnNUOEH3HUA9PwHqAHgEYqAGmkLmAMxIBPMKqJU%2BOmABYjJ01STAAXrep6AzAB0AKwA4q6mMlQgEACuqDCQwAlqAHJofG5MAL7Gpln0BVFKKmpUmoK6AEwRkdQWMNa%2BOhZ5dR7evlQAnP56IeFtkdFxCUkpROmomabZJgC66HM0VJhgOFBIfFhEWMIwEiRQhnBswCQSe5wHRydHcAA%2BcFN8828UiCiogRw4WEgACiQ5Fw%2BAAlEA)

Thank you.

**Comment by KotlinIsland** [comment]
```py
json.loads("1")
```

**Comment by DetachHead** [comment]
`reportAny` is warning you that the type annotation does not guarantee that the type will be what you expect. for example:

```py
import json


data: str = """ {
   "blockdevices": 1
}
"""

parsed: dict[str, list[dict[str, str | None]]] = json.loads(s=data)
```

since there's no runtime validation here, the type annotation is unsafe. if you are certain the parsed data is always in the correct format, you can either use a `cast`:

```py
fromk typing import cast

parsed = cast(dict[str, list[dict[str, str | None]]], json.loads(s=data))
```

or disable the `reportAny` rule either with a `# pyright:ignore[reportAny]` comment, or globally:

```toml
[tool.basedpyright]
reportAny = "none"
```

or if you just want the same type checking rules as pyright:

```toml
[tool.basedpyright]
typeCheckingMode = "standard"
```

**Comment by Sevrain1** [comment]
Thank you very much for the detailed explanation @DetachHead.
I guess in this case since I'm sure the parsed data will always be in this specific format, I should use casting.

Thanks again :-)
